SignalR now uses System.Text.Json (#14902)
aspnet/AspNetCore.Docs: aspnetcore/signalr/configuration.md

ASP.NET Core SignalR supports two protocols for encoding messages: [JSON](https://www.json.org/) and [MessagePack](https://msgpack.org/index.html). Each protocol has serialization configuration options.

> JSON serialization can be configured on the server using the [AddJsonProtocol](/dotnet/api/microsoft.extensions.dependencyinjection.jsonprotocoldependencyinjectionextensions.addjsonprotocol) extension method, which can be added after [AddSignalR](/dotnet/api/microsoft.extensions.dependencyinjection.signalrdependencyinjectionextensions.addsignalr) in your `Startup.ConfigureServices` method. The `AddJsonProtocol` method takes a delegate that receives an `options` object. The [PayloadSerializerOptions](/dotnet/api/microsoft.aspnetcore.signalr.jsonhubprotocoloptions.payloadserializeroptions) property on that object is a System.Text.Json `JsonSerializerOptions` object that can be used to configure serialization of arguments and return values. See the [System.Text.Json Documentation](dotnet/api/system.text.json) for more details.

Reviewer: System.Text.Json Documentation --> System.Text.Json documentation

As an example, to configure the serializer to use "PascalCase" property names, instead of the default "camelCase" names, use the following code:

```csharp
services.AddSignalR()
    .AddJsonProtocol(options => {
        options.PayloadSerializerOptions.PropertyNamingPolicy =

>             JsonNamingPolicy.CamelCase;

Reviewer: The intro text for this says it configures the serializer to use pascal case, but this code configures it to use camel case.

ASP.NET Core SignalR supports two protocols for encoding messages: [JSON](https://www.json.org/) and [MessagePack](https://msgpack.org/index.html). Each protocol has serialization configuration options.

> ::: moniker range=">= aspnetcore-3.0"

Reviewer 2: I wonder if we should be mentioning the `AddNewtonsoftJsonProtocol` extension method in this doc

Reviewer 3: Yes it does

Author: The 2.2-to-3.0 migration doc discusses AddNewtonsoftJsonProtocol, but not this SignalR configuration doc. Should we copy the [Switch to Newtonsoft.Json](https://github.com/aspnet/AspNetCore.Docs/blob/f849554c6000e74e9376d359da20c686673ab880/aspnetcore/migration/22-to-30.md#switch-to-newtonsoftjson) section from the migration doc to this doc?

Reviewer 2: Yeah, it looks pretty good

> JSON serialization can be configured on the server using the [AddJsonProtocol](/dotnet/api/microsoft.extensions.dependencyinjection.jsonprotocoldependencyinjectionextensions.addjsonprotocol) extension method, which can be added after [AddSignalR](/dotnet/api/microsoft.extensions.dependencyinjection.signalrdependencyinjectionextensions.addsignalr) in your `Startup.ConfigureServices` method. The `AddJsonProtocol` method takes a delegate that receives an `options` object. The [PayloadSerializerOptions](/dotnet/api/microsoft.aspnetcore.signalr.jsonhubprotocoloptions.payloadserializeroptions) property on that object is a System.Text.Json `JsonSerializerOptions` object that can be used to configure serialization of arguments and return values. See the [System.Text.Json documentation](dotnet/api/system.text.json) for more details.

Reviewer 4: ```suggestion
JSON serialization can be configured on the server using the [AddJsonProtocol](/dotnet/api/microsoft.extensions.dependencyinjection.jsonprotocoldependencyinjectionextensions.addjsonprotocol) extension method, which can be added after [AddSignalR](/dotnet/api/microsoft.extensions.dependencyinjection.signalrdependencyinjectionextensions.addsignalr) in your `Startup.ConfigureServices` method. The `AddJsonProtocol` method takes a delegate that receives an `options` object. The [PayloadSerializerOptions](/dotnet/api/microsoft.aspnetcore.signalr.jsonhubprotocoloptions.payloadserializeroptions) property on that object is a `System.Text.Json` <xref:System.Text.Json.JsonSerializerOptions> object that can be used to configure serialization of arguments and return values. For more information, see the [System.Text.Json documentation](/dotnet/api/system.text.json).
```

> As an example, to configure the serializer to not change the casing of property names, instead of the default "camelCase" names, use the following code:

Reviewer 4: ```suggestion
As an example, to configure the serializer to not change the casing of property names, instead of the default "camelCase" names, use the following code in `Startup.ConfigureServices`:
```

// At the top of the file:
using Microsoft.Extensions.DependencyInjection;

// When constructing your connection:
var connection = new HubConnectionBuilder()
    .AddJsonProtocol(options => {
        options.PayloadSerializerOptions.PropertyNamingPolicy = null;
    })
    .Build();
```

::: moniker-end

> ::: moniker range="= aspnetcore-2.2"

Reviewer 4: I suspect this content should display for both 2.1 and 2.2 (since this doc is available for 2.1 and later). If that's the case, make the following change:

```suggestion
::: moniker range="<= aspnetcore-2.2"
```

> JSON serialization can be configured on the server using the [AddJsonProtocol](/dotnet/api/microsoft.extensions.dependencyinjection.jsonprotocoldependencyinjectionextensions.addjsonprotocol) extension method, which can be added after [AddSignalR](/dotnet/api/microsoft.extensions.dependencyinjection.signalrdependencyinjectionextensions.addsignalr) in your `Startup.ConfigureServices` method. The `AddJsonProtocol` method takes a delegate that receives an `options` object. The [PayloadSerializerSettings](/dotnet/api/microsoft.aspnetcore.signalr.jsonhubprotocoloptions.payloadserializersettings) property on that object is a JSON.NET `JsonSerializerSettings` object that can be used to configure serialization of arguments and return values. See the [JSON.NET documentation](https://www.newtonsoft.com/json/help/html/Introduction.htm) for more details.

Reviewer 4: ```suggestion
JSON serialization can be configured on the server using the [AddJsonProtocol](/dotnet/api/microsoft.extensions.dependencyinjection.jsonprotocoldependencyinjectionextensions.addjsonprotocol) extension method, which can be added after [AddSignalR](/dotnet/api/microsoft.extensions.dependencyinjection.signalrdependencyinjectionextensions.addsignalr) in your `Startup.ConfigureServices` method. The `AddJsonProtocol` method takes a delegate that receives an `options` object. The [PayloadSerializerSettings](/dotnet/api/microsoft.aspnetcore.signalr.jsonhubprotocoloptions.payloadserializersettings) property on that object is a JSON.NET `JsonSerializerSettings` object that can be used to configure serialization of arguments and return values. For more information, see the [JSON.NET documentation](https://www.newtonsoft.com/json/help/html/Introduction.htm).
```

> As an example, to configure the serializer to use "PascalCase" property names, instead of the default "camelCase" names, use the following code:

Reviewer 4: ```suggestion
As an example, to configure the serializer to use "PascalCase" property names, instead of the default "camelCase" names, use the following code in `Startup.ConfigureServices`:
```
